scripts/irods/paths.py

```diff
@@ -1,139 +1,469 @@
 import grp
 import os
 import pwd
-
+import pathlib
+import logging
+
+from .exceptions import IrodsError
+
+try:
+    from . import paths_cmake
+    no_paths_cmake = False
+except ImportError:
+    no_paths_cmake = True
+
+# If making changes to this file, equivalent changes may also be needed in
+# lib/core/src/irods_default_paths.cpp
+
+# TODO: consider completely transitioning to pathlib paths.
+
+# Returns a normalized pathlib.PurePath of CMake-provided IRODS_HOME_DIRECTORY
+# Not guaranteed to return a usable value. Use _irods_directory() instead.
+_irods_install_directory_cache = None
+def _irods_install_directory():
+    global _irods_install_directory_cache
+    if _irods_install_directory_cache is None:
+        if no_paths_cmake:
+            raise IrodsError('Running from source tree. No install directories.')
+        _irods_install_directory_cache = pathlib.PurePath(os.path.normpath(paths_cmake.cmake_dir_irodshome))
+    return _irods_install_directory_cache
+
+# Returns a normalized pathlib.PurePath of CMake-provided CMAKE_INSTALL_SYSCONFDIR
+# Not guaranteed to return a usable value. Use _sysconf_directory() instead.
```

Comment:
The following functions call their `*_install_*` equivalents, so it feels like there's no way around the situation in which no usable value is returned...
- `_sysconf_directory`
- `_localstate_directory`
- `_plugins_directory`
- `_server_bin_directory`
- `_runstate_directory`

`_irods_directory` does, however, do something, so that comment is still valid. Am I misunderstanding?

Reply:
Those functions call their `*_install_directory()` equivalents to derive usable values from them.
`_irods_directory` does not rely on an `*_install_directory()`, which is how we are able to do this.

It's the same logic as what's in `irods_default_paths.cpp`. The values from the macros are not usable on their own.

Reply:
Okay, I guess I just find the comment here confusing, then:

 > Not guaranteed to return a usable value. Use _sysconf_directory() instead.

We derive a usable value from this function but it's not guaranteed to return a usable value. In that case, we recommend calling `_sysconf_directory()` instead of `_sysconf_install_directory()`, which will just call `_sysconf_install_directory()` anyway. So why are we recommending it as an alternative? Am I overthinking this?

This is not a blocker, I just don't understand what this means (and the equivalent comments for the functions listed above).

Reply:
Gotcha. Thanks!